### PR TITLE
Menu

### PR DESCRIPTION
**Pull request by stped18**
this menu is using libgdx indput processor 
//TODO change libgdx input to the keyindputmodel:
// TODO creating comment models for alle Entitys and validate in the gameScreen 
the input model is not done so when it is done the menu can use the mouse and key handling

### CODE REVIEW

**Review by Kjalris — APPROVED** [review]

**Review by Lucassass — APPROVED** [review]

**Review by DenBedste — APPROVED** [review]

File: dk.sdu.se4.common/src/main/java/dk/sdu/se4/common/entity/part/VisibilityPart.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+import dk.sdu.se4.common.entity.Entity;
+
+/**
+ *
+ * @author steff
+ */
+public class VisibilityPart implements EntityPart{
+
+    private boolean VisibilityPart;
```
Comment (Nicklason) [inline dk.sdu.se4.common/src/main/java/dk/sdu/se4/common/entity/part/VisibilityPart.java:16]: this variable name is confusing

File: dk.sdu.se4.common/src/main/java/dk/sdu/se4/common/service/GameService.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  void removePlugin(PluginService pluginService);
+
+  void addProcessorService(ProcessorService ProcessorService);
+
+  void removeProcessorService(ProcessorService ProcessorService);
+
+  void addPostProcessorService(PostProcessorService postProcessorService);
+
+  void removePostProcessorService(PostProcessorService postProcessorService);
```
Comment (Nicklason) [inline dk.sdu.se4.common/src/main/java/dk/sdu/se4/common/service/GameService.java:31]: why did you add interfaces for these methods?

File: dk.sdu.se4.core/src/main/resources/META-INF/gamescreen.xml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+               name="PluginService" policy="static" unbind="removePlugin"/>
+   
+    
+    <reference bind="addProcessorService" cardinality="0..n"
+               interface="dk.sdu.se4.common.service.ProcessorService"
+               name="ProcessorService" policy="static" unbind="removeProcessorService" />
+    
+    <reference bind="addPostProcessorService" cardinality="0..n"
+               interface="dk.sdu.se4.common.service.PostProcessorService"
+               name="PostProcessorService" policy="static" unbind="removePostProcessorService" />-->
```
Comment (Nicklason) [inline dk.sdu.se4.core/src/main/resources/META-INF/gamescreen.xml:24]: remove comments

**Review by Nicklason — CHANGES_REQUESTED** [review]
I don't like that you use entity for the screens, but also that you have implemented a lot of things that are not part of the MVP. Stick to the plan.

File: dk.sdu.se4.common/src/main/java/dk/sdu/se4/common/entity/part/VisibilityPart.java (hunk 1 above)
Comment (stped18) [reply dk.sdu.se4.common/src/main/java/dk/sdu/se4/common/entity/part/VisibilityPart.java:16]: yes i deleted that

**Comment by stped18** [comment]
> I don't like that you use entity for the screens, but also that you have implemented a lot of things that are not part of the MVP. Stick to the plan.

removed

File: dk.sdu.se4.common/src/main/java/dk/sdu/se4/common/service/GameService.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+import java.util.List;
+
+public interface GameService {
+
+  void create();
+
+  void addMapService(MapService mapService);
+
+  void removeMapService(MapService mapService);
```
Comment (Nicklason) [inline dk.sdu.se4.common/src/main/java/dk/sdu/se4/common/service/GameService.java:17]: Should not be part of the interface

File: dk.sdu.se4.common/src/main/java/dk/sdu/se4/common/service/GameService.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+  SpriteBatch getBatch();
+
+  int getWidth();
+
+  int  getHeight();
+
+ List<ProcessorService> getProcessorServices();
+
+  MapService getMapService();
```
Comment (Nicklason) [inline dk.sdu.se4.common/src/main/java/dk/sdu/se4/common/service/GameService.java:29]: Should not be part of the interface

**Review by Nicklason — CHANGES_REQUESTED** [review]

File: dk.sdu.se4.common/src/main/java/dk/sdu/se4/common/service/GameService.java (hunk 5 above)
Comment (stped18) [reply dk.sdu.se4.common/src/main/java/dk/sdu/se4/common/service/GameService.java:23]: yes

File: dk.sdu.se4.common/src/main/java/dk/sdu/se4/common/service/GameService.java (hunk 4 above)
Comment (stped18) [reply dk.sdu.se4.common/src/main/java/dk/sdu/se4/common/service/GameService.java]: no

Comment (stped18) [reply dk.sdu.se4.common/src/main/java/dk/sdu/se4/common/service/GameService.java]: I remove it again

Comment (stped18) [reply dk.sdu.se4.common/src/main/java/dk/sdu/se4/common/service/GameService.java]: but it cant the interface is the contract of the class and it sai with methods shot bee there

File: dk.sdu.se4.common/src/main/java/dk/sdu/se4/common/service/GameService.java (hunk 5 above)
Comment (stped18) [reply dk.sdu.se4.common/src/main/java/dk/sdu/se4/common/service/GameService.java:23]: it is used in the Screen model

File: dk.sdu.se4.common/src/main/java/dk/sdu/se4/common/service/GameService.java (hunk 4 above)
Comment (stped18) [reply dk.sdu.se4.common/src/main/java/dk/sdu/se4/common/service/GameService.java]: I remove create and add and remove

File: dk.sdu.se4.common/src/main/java/dk/sdu/se4/common/service/GameService.java (hunk 5 above)
Comment (stped18) [reply dk.sdu.se4.common/src/main/java/dk/sdu/se4/common/service/GameService.java:23]: yes i shot

**Review by Nicklason — APPROVED** [review]
